pelican/cache.py

```diff
@@ -1,15 +1,17 @@
+# -*- coding: utf-8 -*-
 from __future__ import unicode_literals
 
 import hashlib
 import logging
 import os
+
+from pelican.utils import mkdir_p
```

Comment:
First, `pep8` tool currently does not check the order of imports, it only makes sure that there are no unneeded imports or double imports;

You are right that the pickle import is not local and should be above, my reasoning for keeping this order is the try block.

When looking at the file you first got the imports, which generally do not contain any logic, but whlie trying to import cpickle we suddenly have logic (if an error occurs, do something different). For me that is something different to a normal import and should be below the normal import blocks.

I was also the original author of this line, so from a coder point of few I say this change was not needed. While reviewing however I felt differently and wanted to move it.

All in all pep8 only defines the order of imports as 'should' and not 'must'. I will however bow to popular demand on this one. (Things like this are the reason I did not just run autopep on it and went home)